avpnc

```diff
@@ -40,25 +57,36 @@ def openvpn(username, password):
         proc.communicate()
 
 
+credentials = {}
+
+
 class Handler(BaseHTTPRequestHandler):
     def log_message(self, format, *args):
         return
 
     def do_POST(self):
         self.send_response(200)
-        self.send_header('Access-Control-Allow-Origin', '*')
+        self.send_header("Access-Control-Allow-Origin", "*")
         self.end_headers()
-        self.wfile.write(b'SuccessAviatrix')
+        self.wfile.write(b"SuccessAviatrix")
         global credentials
         credentials = json.loads(unquote(self.path[1:]))
 
 
-TCPServer.allow_reuse_address = True
-with TCPServer(('localhost', 15395), Handler) as httpd:
-    webbrowser.open(get_url())
-    httpd.handle_request()
+if X_is_running():
+    TCPServer.allow_reuse_address = True
+    with TCPServer(("localhost", 15395), Handler) as httpd:
+        print("opening browser")
+        webbrowser.open(get_url())
+        httpd.handle_request()
+else:
+    print("X is not running")
+    print("open below url in browser and copy Email and Token")
+    print(get_url())
+    credentials["Email"] = input("Enter Email: ")
+    credentials["Token"] = input("Enter Token: ")
```

Comment:
If you already have the email/token, what's your usecase for using this? You could simply start openvpn directly.

Reply:
yes.. you are correct.. :+1: